src/evaluate.c

```diff
@@ -0,0 +1,120 @@
+#include "evaluate.h"
+
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+#ifdef __APPLE__
+#include <sys/_types/_size_t.h>
+#endif
+
+enum Constants{
+  kSizeOfArrayForTesting = 5,
+  kArbitrarySizeOfValuesInTest = 500,
+  kWeightOfPreferences = 25,
+  kHundredPercent = 100
+};
+
+/* This function is included as initial testing code to populate the struct
+   array. */
+void TestingFunction() {
+  srand(time(NULL));
+  size_t size_of_struct_array = kSizeOfArrayForTesting;
+  Trip trip_arr[size_of_struct_array];
+  size_t size_of_trip = sizeof(Trip);
+
+  // Population the struct array.
+  for (size_t i = 0; i < size_of_struct_array; i++) {
+    trip_arr[i].price_ = (double)(rand() % kArbitrarySizeOfValuesInTest);
+    trip_arr[i].comfortability_ =
+        (double)(rand() % kArbitrarySizeOfValuesInTest);
+    trip_arr[i].time_ = (double)(rand() % kArbitrarySizeOfValuesInTest);
+    trip_arr[i].emissions_ = (double)(rand() % kArbitrarySizeOfValuesInTest);
+  }
+  Evaluate(trip_arr, size_of_struct_array, size_of_trip);
+}
+
+void CalculateScore(Trip trips[], size_t num_trips, size_t read_offset,
```

Comment:
@MadsHeilmann Fix this issue properly. Take a look at how the others from the team approached it, otherwise ask for help. Don't just blindly ignore every issue there is